inference-engine/tests/functional/plugin/shared/include/behavior/infer_request.hpp

```diff
@@ -623,4 +625,43 @@ TEST_P(InferRequestTests, returnDeviceBusyOnGetPerformanceCountAfterAsyncInfer)
         std::cout << "Exception" << e.what() << std::endl;
     }
 }
-}  // namespace BehaviorTestsDefinitions
\ No newline at end of file
+
+class InferRequestTestsResultNotReady : public InferRequestTests {
+};
+
+TEST_P(InferRequestTestsResultNotReady, ReturnResultNotReadyFromWaitInAsyncModeForTooSmallTimeout) {
+    // Skip test according to plugin specific disabledTestPatterns() (if any)
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    // Create CNNNetwork from ngraph::Function
+    // return function which computes around 20ms on GNA SW
+    function = LayerTestsDefinitions::Basic_LSTM_S::GetNetwork(3000, 380);
+    InferenceEngine::CNNNetwork cnnNet(function);
+    // Load CNNNetwork to target plugins
+    auto execNet = ie->LoadNetwork(cnnNet, targetDevice, configuration);
+    // Create InferRequest
+    InferenceEngine::InferRequest req;
+    ASSERT_NO_THROW(req = execNet.CreateInferRequest());
+    InferenceEngine::ResponseDesc response;
+    InferenceEngine::StatusCode sts = InferenceEngine::StatusCode::OK;
+    std::chrono::system_clock::time_point callbackTimeStamp;
+    volatile bool callbackTimeStampSet = false;
```

Comment:
Please use :
```c++ 
std::promise<std::chrono::system_clock::time_point> 
std::future<std::chrono::system_clock::time_point> 
```
instead of value and volatile flag.

Reply:
Done